collector/lib/NetworkStatusNotifier.cpp

```diff
@@ -240,12 +242,19 @@ void NetworkStatusNotifier::RunSingle(IDuplexClientWriter<sensor::NetworkConnect
     const sensor::NetworkConnectionInfoMessage* msg;
     ConnMap new_conn_state, delta_conn;
     AdvertisedEndpointMap new_cep_state;
+    bool enableExternalIPs = config_.EnableExternalIPs();
+
     WITH_TIMER(CollectorStats::net_fetch_state) {
-      conn_tracker_->EnableExternalIPs(config_.EnableExternalIPs());
+      conn_tracker_->EnableExternalIPs(enableExternalIPs);
 
       new_conn_state = conn_tracker_->FetchConnState(true, true);
       if (enable_afterglow_) {
         ConnectionTracker::ComputeDeltaAfterglow(new_conn_state, old_conn_state, delta_conn, time_micros, time_at_last_scrape, afterglow_period_micros_);
+        if (!prevEnableExternalIPs && enableExternalIPs) {
+          ConnectionTracker::CloseNormalizedConnections(&old_conn_state, &delta_conn);
+        } else if (prevEnableExternalIPs && !enableExternalIPs) {
+          conn_tracker_->CloseExternalUnnormalizedConnections(&old_conn_state, &delta_conn);
+        }
```

Comment:
I think it would make more sense to move this block into `ConnectionTracker::EnableExternalIPs()`, since it only affects `conn_tracker_` and it already has the previous and new state of the variable in it, so we don't need to modify `NetworkStatusNotifier` at all.

Reply:
After looking a bit closer, there's no easy way to move the code into conn_tracker_, the whole block is a tangly mess so I'll let it slide for now, but we need to take a closer look at this entire logic and figure out a sensible way to untangle it.